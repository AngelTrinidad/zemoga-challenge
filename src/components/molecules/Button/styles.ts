import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  buttonText: {
    color: 'white',
  },
});

export default styles;
