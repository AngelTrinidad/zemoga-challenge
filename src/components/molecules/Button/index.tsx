import React, {FC} from 'react';

import {TextStyle} from 'react-native/types';

import styles from './styles';

import BasicButton, {ButtonProps} from '~components/atoms/BasicButton';
import Typography, {TypographyVariant} from '~components/atoms/Typography';

interface Props extends ButtonProps {
  title: string;
  textStyle?: TextStyle;
  titleVariant?: TypographyVariant;
}

const Button: FC<Props> = ({
  title,
  textStyle,
  titleVariant = 'lg',
  ...props
}) => {
  return (
    <BasicButton {...props}>
      <Typography variant={titleVariant} style={[styles.buttonText, textStyle]}>
        {title}
      </Typography>
    </BasicButton>
  );
};

export default Button;
