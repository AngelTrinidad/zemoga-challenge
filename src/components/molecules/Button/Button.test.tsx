import React from 'react';

import {fireEvent, render} from '@testing-library/react-native';

import Button from '.';

describe('<Button />', () => {
  let component: any;
  const onPressMock = jest.fn();

  beforeEach(() => {
    component = render(<Button onPress={onPressMock} title="Press" />);
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call onPress callback on tapped it', () => {
    const button = component.getByText('Press');
    fireEvent(button, 'press');
    expect(onPressMock).toBeCalled();
  });
});
