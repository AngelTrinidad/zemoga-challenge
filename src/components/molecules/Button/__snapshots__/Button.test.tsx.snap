// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should has same snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 10,
        "height": 57.14285714285714,
        "justifyContent": "center",
        "width": "100%",
      },
      {
        "backgroundColor": "#334FFA",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "Metropolis Bold",
        "fontSize": 19.65714285714286,
        "fontWeight": "800",
      }
    }
  >
    Press
  </Text>
</View>
`;
